### PR TITLE
Fixing warnings in CI compilation

### PR DESCRIPTION
**Pull request by lixiumei123**

### CODE REVIEW

File: src/utils/neu_jwt.c [hunk 1]
```diff
@@ -107,7 +107,7 @@ int neu_jwt_new(char **token)
     }
 
     *token = str_jwt;
-    strncpy(token_local, *token, strlen(*token));
+    memcpy(token_local, *token, strlen(*token));
```
Comment (fengzeroz) [inline src/utils/neu_jwt.c:110]: which warning

File: src/event/event_linux.c [hunk 2]
```diff
@@ -95,7 +96,10 @@ static void *event_loop(void *arg)
             if ((event.events & EPOLLIN) == EPOLLIN) {
                 uint64_t t;
 
-                read(data->fd, &t, sizeof(t));
+                ssize_t size = read(data->fd, &t, sizeof(t));
+                if (size < 0) {
+                    assert(size);
+                }
```
Comment (fengzeroz) [inline src/event/event_linux.c:102]: how to use assert

File: src/daemon.c [hunk 3]
```diff
@@ -111,8 +118,16 @@ int neuron_already_running()
...
+    if (ret < 0) {
+        assert(ret);
+    }
+
     snprintf(buf, sizeof(buf), "%ld", (long) getpid());
-    write(fd, buf, strlen(buf) + 1);
+    size = write(fd, buf, strlen(buf) + 1);
+    if (size < 0) {
+        assert(size);
+    }
```
Comment (fengzeroz) [inline src/daemon.c:130]: how to use assert

File: src/daemon.c [hunk 4]
```diff
@@ -75,8 +77,11 @@ void daemonize()
 
     // attach file descriptors 0, 1 and 2 to /dev/null
     open("/dev/null", O_RDWR);
-    dup(0);
-    dup(0);
+    ret = dup(0);
+    ret = dup(0);
+    if (ret < 0) {
+        assert(ret);
+    }
```
Comment (fengzeroz) [inline src/daemon.c:84]: return value of the first call is overwritten

File: src/base/tag_table.c [hunk 5]
```diff
@@ -113,7 +113,7 @@ int neu_tag_table_add(neu_tag_table_t *tbl, neu_tag_t *tag)
         elem = calloc(1, sizeof(struct tag_elem));
 
         elem->tag = *tag;
-        strncpy(elem->name, tag->name, strlen(tag->name));
+        strncpy(elem->name, tag->name, sizeof(tag->name));
```
Comment (fengzeroz) [inline src/base/tag_table.c:116]: what type of tag name is

File: src/base/neu_tag.c [hunk 6]
```diff
@@ -426,7 +426,7 @@ int neu_datatag_string_dtoh(char *str, int len)
         t[i] = str[i * 2 + 1];
     }
     memset(str, 0, len);
-    strncpy(str, t, strlen(t));
+    memcpy(str, t, strlen(t));
```
Comment (fengzeroz) [inline src/base/neu_tag.c:429]: which kind of warning

File: src/base/neu_tag.c [hunk 7]
```diff
@@ -412,7 +412,7 @@ int neu_datatag_string_etoh(char *str, int len)
         t[i] = str[i * 2];
     }
     memset(str, 0, len);
-    strncpy(str, t, strlen(t));
+    memcpy(str, t, strlen(t));
```
Comment (fengzeroz) [inline src/base/neu_tag.c:415]: which kind of warning

File: src/base/neu_tag.c (hunk 6 above)
Comment (lixiumei123) [reply src/base/neu_tag.c:429]: 截屏2022-04-15 上午9 53 20

File: src/base/neu_tag.c (hunk 7 above)
Comment (lixiumei123) [reply src/base/neu_tag.c:415]: 截屏2022-04-15 上午9 56 49

**Pull request merged**